### PR TITLE
workaround msys2 bug

### PR DESCRIPTION
**Pull request by mati865**

### CODE REVIEW

**Comment by mati865** [comment]
The worst thing that can happen is different error, so...

File: src/ci/azure-pipelines/steps/run.yml [hunk 1]
```diff
@@ -42,6 +42,8 @@ steps:
       fi
     displayName: Decide whether to run this job
 
+repo.msys2.org/mingw/i686/mingw-w64-i686-ca-certificates-20180409-1-any.pkg.tar.xz
+repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20180409-1-any.pkg.tar.xz
```
Comment (pietroalbini) [inline src/ci/azure-pipelines/steps/run.yml]: What are those two links?

File: src/ci/azure-pipelines/steps/install-windows-build-deps.yml [hunk 2]
```diff
@@ -84,6 +84,11 @@ steps:
   condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'), ne(variables['MINGW_URL'],''))
   displayName: Download custom MinGW
 
+#workaround msys bug, step 1
+- bash: |
+    curl repo.msys2.org/mingw/i686/mingw-w64-i686-ca-certificates-20180409-1-any.pkg.tar.xz -O
```
Comment (pietroalbini) [inline src/ci/azure-pipelines/steps/install-windows-build-deps.yml]:
```suggestion
    curl https://repo.msys2.org/mingw/i686/mingw-w64-i686-ca-certificates-20180409-1-any.pkg.tar.xz -O
```

File: src/ci/azure-pipelines/steps/install-windows-build-deps.yml [hunk 3]
```diff
@@ -84,6 +84,11 @@ steps:
...
+    curl repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20180409-1-any.pkg.tar.xz -O
```
Comment (pietroalbini) [inline src/ci/azure-pipelines/steps/install-windows-build-deps.yml]:
```suggestion
    curl https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20180409-1-any.pkg.tar.xz -O
```

**Review by emilyalbini — CHANGES_REQUESTED** [review]

File: src/ci/azure-pipelines/steps/run.yml (hunk 1 above)
Comment (mati865) [reply src/ci/azure-pipelines/steps/run.yml]: Oh I hate notepad

**Comment by pietroalbini** [comment]
You also forgot the `condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))` on both task.

**Comment by mati865** [comment]
I'm bad at it...

@bors r=pietroalbini

**Comment by mati865** [comment]
Probably won't work for me

@bors treeclosed-

**Comment by Mark-Simulacrum** [comment]
I've included this into beta in https://github.com/rust-lang/rust/pull/65708 but nominated for infra just to check in that that's fine (the beta rollup will hopefully have landed by then). It's a pure CI change that shouldn't affect distributed code so I'm not too concerned.

**Pull request merged**